packages/framer-motion/src/components/AnimatePresence/index.tsx

```diff
@@ -96,17 +96,14 @@ export const AnimatePresence: React.FunctionComponent<
     const isMounted = useIsMounted()
 
     // Filter out any children that aren't ReactElements. We can only track ReactElements with a props.key
-    const filteredChildren = onlyElements(children)
-    let childrenToRender = filteredChildren
+    const filteredChildren = useRef(onlyElements(children))
+    filteredChildren.current = onlyElements(children)
+    let childrenToRender = filteredChildren.current
 
     const exitingChildren = useRef(
         new Map<ComponentKey, ReactElement<any> | undefined>()
     ).current
 
-    // Keep a living record of the children we're actually rendering so we
-    // can diff to figure out which are entering and exiting
-    const presentChildren = useRef(childrenToRender)
```

Comment:
Removed `presentChildren` for clarity - `filteredChildren`, `exitingChildren` and `allChildren` are enough to maintain the status quo